src/VisualStudio/IntegrationTest/New.IntegrationTests/CSharp/CSharpCodeActions.cs

```diff
@@ -1458,5 +1458,61 @@ public async Task TestNonSourceDocumentRefactoring()
 
             AssertEx.EqualOrDiff(expectedText, await TestServices.Editor.GetTextAsync(HangMitigatingCancellationToken));
         }
+
+        [IdeFact, Trait(Traits.Feature, Traits.Features.CodeGeneration)]
+        public async Task TestRefactoringsAreSortedByPriority()
+        {
+            await SetUpEditorAsync(@"
+#pragma warning disable IDE0060 // Remove unused parameter
+class C
+{ 
+    public C(int x1, int $$x2, int x3)
+    {
+    }
+}", HangMitigatingCancellationToken);
+
+            await TestServices.Editor.InvokeCodeActionListAsync(HangMitigatingCancellationToken);
+
+            var expectedItems = new[]
+            {
+                "Create and assign property 'X2'",
+                "Create and assign field 'x2'",
+                "Create and assign remaining as properties",
+                "Create and assign remaining as fields",
+                "Change signature...",
+                "Wrap every parameter",
+                "Align wrapped parameters",
+                "Indent all parameters",
+                "Indent wrapped parameters",
+                "Unwrap and indent all parameters",
+            };
+
+            await TestServices.EditorVerifier.CodeActionsAsync(expectedItems, ensureExpectedItemsAreOrdered: true, cancellationToken: HangMitigatingCancellationToken);
+
+            await SetUpEditorAsync(@"
```

Comment:
Added 2 invocations as it seems the first one might succeed often, but for 9 out of 10 times I saw either first or second one has unexpected order of the items.